django/db/models/sql/compiler.py

```diff
@@ -125,11 +125,12 @@ def get_group_by(self, select, order_by):
             cols = expr.get_group_by_cols()
             for col in cols:
                 expressions.append(col)
-        for expr, (sql, params, is_ref) in order_by:
-            # Skip References to the select clause, as all expressions in the
-            # select clause are already part of the group by.
-            if not is_ref:
-                expressions.extend(expr.get_group_by_cols())
+        if not self._meta_ordering:
+            for expr, (sql, params, is_ref) in order_by:
```

Comment:
I don't think this is completely correct. Models with a `Meta.ordering` can still make use of explicit `order_by` and when it's the case it should be honored.

Reply:
You are right - there is similar issue in case we using explicit ordering by related model. I will try to fix it also.

Reply:
@charettes 
While I agree, that explicit ordering produces redundand GROUP BY field, I beleave it doesn't affects query result.
Do you think it does affects query result? (I just can not create functional unit test case for it)

Reply:
@charettes We set `_meta_ordering` as a last resort:
https://github.com/django/django/blob/50ef9dfac21499ff6c46c2f6a33103e40b60cd4e/django/db/models/sql/compiler.py#L280-L290
so this cannot affect queries with explicit `order_by()` calls.

Maybe a short comment:
```
# Default ordering doesn't affect the GROUP BY clause.
```

Reply:
Default ordering of related models does affects `GROUP BY ` clause, but it doesn't matter for query result as I think

https://github.com/django/django/blob/50ef9dfac21499ff6c46c2f6a33103e40b60cd4e/django/db/models/sql/compiler.py#L736-L744

Reply:
I'm confused. How it affects the `GROUP BY` clause with this change?

Reply:
Look to test added by this PR. Let's add `order_by` to our articles query:

```python
articles = Article.objects.values('author').annotate(count=Count('author')).order_by('author')
```

The resulting query will be:
```sql
SELECT "ordering_article"."author_id", COUNT("ordering_article"."author_id") AS "count"
FROM "ordering_article"
LEFT OUTER JOIN "ordering_author" ON ("ordering_article"."author_id" = "ordering_author"."id")
GROUP BY "ordering_article"."author_id", "ordering_author"."id"
ORDER BY "ordering_author"."id" DESC
```

Look to redundand `"ordering_author"."id"` in `GROUP BY` - this field comes from default ordering of Author model:
https://github.com/django/django/blob/50ef9dfac21499ff6c46c2f6a33103e40b60cd4e/tests/ordering/models.py#L23-L24

Reply:
@felixxm thanks for the extra context, I misread `self._meta_ordering` as `self._meta.ordering`.


@nE0sIghT You want to use `order_by('author_id')` and not `order_by('author')` to achieve that [as documented](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#django.db.models.query.QuerySet.order_by).

Reply:
I see. Thanks for explanation

Reply:
> @felixxm thanks for the extra context, I misread `self._meta_ordering` as `self._meta.ordering`.

I did the same at the beginning.